tests/application_development/code_relocation/test_lib/test_lib1.c

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2022 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/sys/printk.h>
+#include <zephyr/ztest.h>
+
+__in_section(data, sram2, var) uint32_t var_lib1_sram2_data = 10U;
```

Comment:
@tari When testing these changes, I noticed that these directives can mask errors in the test (on Cortex-M). Since this directive moves symbols to `data.sram2.var` unconditionally, these variables will still be relocated if `zephyr_code_relocate` is not called on this file, so long as another file in the build is relocated to `sram2`, since then the section name will be defined.

I'm interested to know what other approaches we could use here (particularly if `volatile` would prevent these variables from being moved to RODATA). The other option I can see is creating new RAM sections for each file (SRAM2, SRAM3, SRAM4, etc), so that if the file isn't relocated the required linker section won't be present.

Reply:
I don't think `volatile` would be reliable; it would just enforce that reads always actually read without asserting anything about writability. Creating different sections for each file seems messy but reasonable to me: this is a test so it seems okay if it needs to do some weird things in order to verify behavior that's somewhat complex.